spec/javascript/components/atoms/Button.test.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import Button from '~/components/atoms/Button'
+
+describe('Button', () => {
+  test('it renders a button with the correct default properties', () => {
+    const wrapper = shallow(<Button text='This is a button' />)
+    const selector = 'button'
+    expect(wrapper.find(selector).exists()).toBeTruthy()
+    expect(wrapper.find({ type: 'button' })).toHaveLength(1)
+    expect(wrapper.find(selector).hasClass('margin-bottom-none')).toBeFalsy()
+  })
+
+  test('it renders a button with the correct properties', () => {
+    const selector = 'button[type="submit"]'
```

Comment:
In general, prefer to keep selectors really simple and lean on enzyme's utilities to select specific children.

also, instead of storing the selector and doing wrapper.find(selector) in each `expect`, you can store the child wrapper like:

```
const wrapper = shallow(<Button ... />)

const buttonWrapper = findByNameAndProps(wrapper, 'button', { type: 'submit' })

expect(buttonWrapper)...
expect(buttonWrapper)...
expect(buttonWrapper)...
```

That example also uses the findByNameAndProps util I added, but you can just as easily leave out the type in the selector and just add an extra expect to check the type

Reply:
also helpful, thanks!